agent/agent.go

```diff
@@ -106,17 +69,32 @@ func OpenDatabase(file string) (operator.Database, error) {
 }
 
 // NewLogAgent creates a new carbon log agent.
-func NewLogAgent(cfg *Config, logger *zap.SugaredLogger, operatorDir, databaseFile string) *LogAgent {
-	return &LogAgent{
-		Config:        cfg,
-		SugaredLogger: logger,
-		PluginDir:     operatorDir,
-		Database:      databaseFile,
-		buildParams:   make(map[string]interface{}),
+func NewLogAgent(cfg *Config, logger *zap.SugaredLogger, pluginDir, databaseFile string, buildParams map[string]interface{}) (*LogAgent, error) {
+	database, err := OpenDatabase(databaseFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "open database")
 	}
-}
 
-func (a *LogAgent) WithBuildParameter(key string, value interface{}) *LogAgent {
```

Comment:
Didn't like the builder method?

Reply:
I actually really did, but now that the pipeline is being built inside `NewLogAgent`, we can't add build parameters after the fact. If we wanted, we could make a new `LogAgentBuilder` struct and do something like
```go
agent := NewLogAgentBuilder(&mockCfg, mockLogger, mockPluginDir, mockDatabaseFile).
    WithBuildParameter(mockParameterKey, mockParameterValue).
    Build()
```

This seemed like more overhead than it was worth though when we would only have the one optional parameter. Let me know what you think

Reply:
Makes sense. Thanks for breaking that down.